bitstamp.go

```diff
@@ -202,35 +199,31 @@ func (api *Api) GetTradesParams(symbol string, interval string) (trades []Trade,
 // SubscribeOrderBook subscribes for websocket events and sends order book updates
 // into dataChan. To stop processing, sent to, or close stopChan.
 func (api *Api) SubscribeOrderBook(symb string, dataChan chan<- OrderBook, stopChan <-chan struct{}) error {
-	pusherClient, err := pusher.NewClient(APP_KEY)
+	c, err := NewWsClient()
 	if err != nil {
-		return errors.Wrap(err, "client creation error")
-	}
-	channel := "order_book"
-	if symb != "btcusd" {
-		channel += "_" + symb
-	}
-	err = pusherClient.Subscribe(channel)
-	if err != nil {
-		return errors.Wrap(err, "subscription error")
-	}
-	dataChannelTrade, err := pusherClient.Bind("data")
-	if err != nil {
-		return errors.Wrap(err, "bind error")
+		return errors.Wrap(err, "error initializing client")
 	}
+
+	c.Subscribe(fmt.Sprintf("order_book_%s", symb))
+
 	for {
 		select {
-		case dataEvt, ok := <-dataChannelTrade:
-			if !ok {
-				return errors.New("websocket has been closed")
-			}
-			if ob, err := api.parseOrderBook([]byte(dataEvt.Data)); err == nil {
-				dataChan <- *ob
+		case ev := <-c.Stream:
+			if ev.Event == "data" {
+				b, err := json.Marshal(ev.Data)
```

Comment:
this is weird. first, we unmarshal json in`websocket.go`, then, marshal it again, and then unmarshal into an order book.
can we use json.RawMessage?

Reply:
yes, it will be better